### PR TITLE
restart_process: fix restart_process for images that don't have sh or touch

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/distroless:

8d4dd35e0da3a8bcee65ca7c5a271b6405241076 (2020-10-30 16:21:39 -0400)
restart_process: fix restart_process for images that don't have sh or touch

03b208c71f1979443737788a1bec2bb2a0b5520f (2020-10-30 15:46:49 -0400)
restart_process: add custom_build_with_restart

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
@maiamcc ping! not a big deal, but it does affect some of the more exotic builders (bazel, ko)

File: restart_process/Tiltfile [hunk 1]
```diff
@@ -40,7 +40,10 @@ def _helper(base_ref, ref, entrypoint, live_update, restart_file=RESTART_FILE, *
     # `tilt-restart-wrapper` makes use of `entr` (https://github.com/eradman/entr/) to
     # re-execute $entrypoint whenever $restart_file changes
     if type(entrypoint) == type(""):
-        entrypoint_with_entr = ["/tilt-restart-wrapper", "--watch_file={}".format(restart_file), "sh", "-c", entrypoint]
+        if " " in entrypoint:
+            entrypoint_with_entr = ["/tilt-restart-wrapper", "--watch_file={}".format(restart_file), "sh", "-c", entrypoint]
+        else:
+            entrypoint_with_entr = ["/tilt-restart-wrapper", "--watch_file={}".format(restart_file), entrypoint]
```
Comment (maiamcc) [inline restart_process/Tiltfile:46]: why this change? I'm hesitant to make it even harder for a user to predict how their command will be run.

File: restart_process/Tiltfile [hunk 2]
```diff
@@ -27,11 +27,11 @@ def _helper(base_ref, ref, entrypoint, live_update, restart_file=RESTART_FILE, *
     # declare a new docker build that adds a static binary of tilt-restart-wrapper
     # (which makes use of `entr` to watch files and restart processes) to the user's image
     df = '''
-    FROM tiltdev/restart-helper:2020-10-16 as restart-helper
+    FROM tiltdev/restart-helper:2020-10-30 as restart-helper
 
     FROM {}
     USER root
-    RUN ["touch", "{}"]
+    COPY --from=restart-helper /.restart-proc {}
```
Comment (maiamcc) [inline restart_process/Tiltfile:34]: nice!

**Review by maiamcc — APPROVED** [review]
oh whoops sorry I missed this!

File: restart_process/Tiltfile (hunk 1 above)
Comment (nicks) [reply restart_process/Tiltfile:46]: per the pr description: so that it works even if you don't have `sh` in the container

Comment (maiamcc) [reply restart_process/Tiltfile:46]: doesn't `live_update + [run('date > {}'.format(restart_file))]` require `sh` though?

**Review by maiamcc — APPROVED** [review]

File: restart_process/Tiltfile (hunk 1 above)
Comment (nicks) [reply restart_process/Tiltfile:46]: oof, ya, something was messed up in my test env...maybe we should use a different approach, and just have an assertion that 'sh' and 'date' and 'touch' are all in the container...

Comment (maiamcc) [reply restart_process/Tiltfile:46]: :( yeahhh. or write our own binary to write some stuff to the specified file, that we can pull into the container alongside restart_process_wrapper? :-/

**Comment by nicks** [comment]
ya, let me withdraw this and think thru it some more. we figured out how to make sure Bazel sets a base image with the tools we need, so it's less urgent now
